commit b1bc22322842b1a6fd508ce71898d145f5c658a5
Author: Michael Still <mikal@stillhq.com>
Date:   Wed May 7 06:12:09 2025 +1000

    Implement container image build for kerbside.

    I am fairly sure this will need improvement, but its a starting
    point at least.

diff --git a/docker/kerbside/kerbside/Dockerfile.j2 b/docker/kerbside/kerbside/Dockerfile.j2
new file mode 100644
index 0000000..353a286
--- /dev/null
+++ b/docker/kerbside/kerbside/Dockerfile.j2
@@ -0,0 +1,42 @@
+FROM {{ namespace }}/{{ image_prefix }}openstack-base:{{ tag }}
+{% block labels %}
+LABEL maintainer="{{ maintainer }}" name="{{ image_name }}" build-date="{{ build_date }}"
+{% endblock %}
+
+{% block kerbside_header %}{% endblock %}
+
+{% import "macros.j2" as macros with context %}
+{{ macros.configure_user(name='kerbside', groups='kerbside') }}
+
+{% if base_package_type == 'rpm' %}
+    {% set kerbside_packages = [
+        'mariadb-devel',
+        'pkgconfig'
+    ] %}
+{% elif base_package_type == 'deb' %}
+    {% set kerbside_packages = [
+        'libmariadb-dev-compat',
+        'pkg-config'
+    ] %}
+{% endif %}
+{{ macros.install_packages(kerbside_packages | customizable("packages")) }}
+
+# We currently carry a customized version of openstacksdk until our changes are
+# merged upstream. This is handled as a plugin for now. This is also how you
+# would add support to kerbside for other clouds like Shaken Fist or oVirt.
+ADD plugins-archive /
+RUN pip3 --no-cache-dir install /plugins/*
+
+{% set kerbside_pip_packages = [
+    '/kerbside'
+] %}
+
+ADD kerbside-archive /kerbside-source
+RUN ln -s kerbside-source/* kerbside \
+    && {{ macros.install_pip(kerbside_pip_packages | customizable("pip_packages")) }} \
+    && rm -f /etc/machine-id
+
+{% block kerbside_footer %}{% endblock %}
+{% block footer %}{% endblock %}
+
+USER kerbside
diff --git a/kolla/common/config.py b/kolla/common/config.py
index 678b0db..374945b 100644
--- a/kolla/common/config.py
+++ b/kolla/common/config.py
@@ -93,6 +93,7 @@ _PROFILE_OPTS = [
                     'influxdb',
                     'ironic',
                     'kafka',
+                    'kerbside',
                     'kuryr',
                     'magnum',
                     'manila',
diff --git a/kolla/common/sources.py b/kolla/common/sources.py
index e051467..0715493 100644
--- a/kolla/common/sources.py
+++ b/kolla/common/sources.py
@@ -164,6 +164,10 @@ SOURCES = {
         'type': 'url',
         'location': ('$tarballs_base/openstack/keystone/'
                      'keystone-${openstack_branch}.tar.gz')},
+    'kerbside': {
+        'type': 'url',
+        'location': ('https://shakenfist.com/kerbside/releases/'
+                     'kerbside-${openstack_branch}.tar.gz')},
     'kuryr-base': {
         'type': 'url',
         'location': ('$tarballs_base/openstack/kuryr/'
diff --git a/kolla/common/users.py b/kolla/common/users.py
index 9468396..5fa49e5 100644
--- a/kolla/common/users.py
+++ b/kolla/common/users.py
@@ -345,5 +345,9 @@ USERS = {
     'hsmusers-user': {
         'uid': 42493,  # This is not used, but the group ID is required.
         'gid': 42493,
+    },
+    'kerbside-user': {
+        'uid': 42494,
+        'gid': 42494,
     }
 }
